### PR TITLE
Link Nx functions to Nx.Defn.Kernel counterpart

### PR DESCRIPTION
**Pull request by whatyouhide**
I just linked one to see if this is something we wanna do. If this looks like the right track, I can do a bunch more functions!

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -3276,6 +3276,9 @@ defmodule Nx do
   It will broadcast tensors whenever the dimensions do
   not match and broadcasting is possible.
 
+  If you're using `Nx.Defn.defn/2`, you can use the `Nx.Defn.Kernel.*/2`
+  operator in place of this function.
```
[the remark this answers is not in the document]
Comment (whatyouhide) [reply nx/lib/nx.ex:3280]: @josevalim we're fine with the fact that that's not gonna link? (I am)

[the remark this answers is not in the document]
Comment (josevalim) [reply nx/lib/nx.ex:3280]: I am too! We can go ahead with this approach then. :)

**Comment by jonatanklosko** [comment]
On a related note. Would it make sense to document `use Nx.Defn.Kernel`? When working on a new function it's useful to try things outside `defn` (say, in Livebook) and it's a bit less productive to use `Nx.*` functions and only later convert them to operators when moving to `defn`.

**Comment by josevalim** [comment]
@jonatanklosko my main concern about `use Nx.Defn.Kernel` is that you can't easily rollback to the "previous state", where `*` is not overridden and so on. :(

**Comment by jonatanklosko** [comment]
@josevalim true, in Livebook there's always an option to use a branching section to limit the scope, but I see how this can be problematic. But then, not sure about article-like notebooks, but it's definitely good to know when prototyping, hard call.

**Comment by jonatanklosko** [comment]
Theoretically we could add support for something like `use Nx.Defn.Kernel, :revert`, but that's not so pretty.

**Comment by josevalim** [comment]
@jonatanklosko we could have something like:

```elixir
nx do
  a + b
end
```

But then in order for `grad` to work we need expressions so we would need inputs to be given:


```elixir
nx [a, b] do
  a + b
end
```

and then I am not sure if it is worth it.

**Comment by jonatanklosko** [comment]
@josevalim in Livebook context, limiting to a single block doesn't entirely solve it, because then we are limited to a single cell :<

**Comment by whatyouhide** [comment]
@josevalim alright, I added a bunch of docs!

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**